llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -291,6 +291,53 @@ InstructionCost VPRecipeBase::computeCost(ElementCount VF,
   llvm_unreachable("subclasses should implement computeCost");
 }
 
+InstructionCost
+VPPartialReductionRecipe::computeCost(ElementCount VF,
+                                      VPCostContext &Ctx) const {
+  auto *BinOp = cast<BinaryOperator>(getOperand(0)->getUnderlyingValue());
+  auto *PhiR = cast<VPReductionPHIRecipe>(getOperand(1)->getDefiningRecipe());
+  auto *Phi = cast<PHINode>(PhiR->getUnderlyingValue());
+  auto *ExtA = cast<Instruction>(BinOp->getOperand(0));
+  auto *ExtB = cast<Instruction>(BinOp->getOperand(1));
+  Value *A = ExtA->getOperand(0);
+  return Ctx.TTI.getPartialReductionCost(
```

Comment:
Can the cost be computed by avoiding to rely in the underlying IR instructions, e.g. types should be determined by VPTypeAnalysis, extend kinds from the extend recipes. VPlan simplifications may change the recipe's operands, meaning the underlying IR isn't accurate any more.

Reply:
Still pending?

Reply:
Thanks for the update!